### DISCUSSION TITLE
What are props in React?

### DISCUSSION THREAD

**Discussion by fcfcnb** [drknzz/GitHub-Achievements#278] [Ideas]
Props

**Comment by AamirMir1** [comment]
In ReactJS, the props are a type of object where the value of attributes of a tag is stored. The word “props” implies “properties”, and its working functionality is quite similar to HTML attributes. Basically, these props components are read-only components

**Comment by mesheryar** [comment]
Props (short for properties) in React are a way to pass data from a parent component to a child component. They are read-only and help to make components reusable and modular. Components can receive props as arguments and use them to render dynamic content or behavior.

**Comment by SanjaRogovic** [comment]
React components use props to communicate with each other, information that you pass to a JSX tag - thats it basically. You can pass props from a parent component to a child only.  Information you can pass is any JavaScript value including objects, arrays, and functions. 

Here is a link to an official React docs: https://react.dev/

**Comment by viniarski** [comment]
In React, "props" (short for properties) are a way to pass data from a parent component to a child component. Props are immutable, meaning that they cannot be modified by the child component. They are passed down from parent to child as a read-only attribute.

Props are typically used to customize and configure a child component, allowing the parent component to control the behavior and appearance of its children. They are specified as attributes on a JSX element when the child component is rendered within the parent component.

Here's a simple example:

```jsx
// ParentComponent.js
import React from 'react';
import ChildComponent from './ChildComponent';

function ParentComponent() {
  return (
    <div>
      {/* Passing prop named "message" to ChildComponent */}
      <ChildComponent message="Hello from ParentComponent!" />
    </div>
  );
}

export default ParentComponent;
```

```jsx
// ChildComponent.js
import React from 'react';

function ChildComponent(props) {
  return (
    <div>
      {/* Accessing the "message" prop passed down from ParentComponent */}
      <p>{props.message}</p>
    </div>
  );
}

export default ChildComponent;
```

In this example, the `ParentComponent` passes a prop named "message" to the `ChildComponent`. The `ChildComponent` then receives this prop via its `props` parameter and can render it within its JSX.

**Comment by Saurabh-S-Patil** [comment]
Props (short for properties) in React are used for passing data from a parent component to a child component. They are immutable and are passed down the component tree. Props allow components to be dynamic and reusable by providing a way to customize their behavior and appearance based on external data.

**Comment by HosseinMazhar** [comment]
components are functions which return a jsx and props are nothing but arguments of those functions. you call a component with a prop so you can use those prop in that component to create something.
